src/main/java/seedu/connectus/logic/parser/SearchCommandParser.java

```diff
@@ -19,15 +33,116 @@ public class SearchCommandParser implements Parser<SearchCommand> {
+            }
+            predicate.setInstagram(argMultimap.getValue(PREFIX_SOCMED_INSTAGRAM).get());
+        }
+        if (argMultimap.getValue(PREFIX_SOCMED_TELEGRAM).isPresent()) {
+            if (argMultimap.getValue(PREFIX_SOCMED_TELEGRAM).get().trim().equals("")) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+            }
+            predicate.setTelegram(argMultimap.getValue(PREFIX_SOCMED_TELEGRAM).get());
+        }
+        if (argMultimap.getValue(PREFIX_SOCMED_WHATSAPP).isPresent()) {
+            if (argMultimap.getValue(PREFIX_SOCMED_WHATSAPP).get().trim().equals("")) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+            }
+            predicate.setWhatsapp(argMultimap.getValue(PREFIX_SOCMED_WHATSAPP).get());
+        }
+        Set<String> remarks = new HashSet<>();
+        for (String s : argMultimap.getAllValues(PREFIX_REMARK)) {
+            if (s.trim().equals("")) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+            } else {
+                remarks.add(s);
+            }
+        }
+        if (remarks.size() > 0) {
+            predicate.setRemarks(remarks);
+        }
+        Set<String> modules = new HashSet<>();
+        for (String s : argMultimap.getAllValues(PREFIX_MODULE)) {
+            if (s.trim().equals("")) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+            } else {
+                modules.add(s);
+            }
+        }
+        if (modules.size() > 0) {
+            predicate.setModules(modules);
+        }
+        Set<String> ccas = new HashSet<>();
+        for (String s : argMultimap.getAllValues(PREFIX_CCA)) {
+            if (s.trim().equals("")) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+            } else {
+                ccas.add(s);
+            }
+        }
+        if (ccas.size() > 0) {
+            predicate.setCcas(ccas);
+        }
+        Set<String> ccaPositions = new HashSet<>();
+        for (String s : argMultimap.getAllValues(PREFIX_CCA_POSITION)) {
+            if (s.trim().equals("")) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+            } else {
+                ccaPositions.add(s);
+            }
+        }
+        if (ccaPositions.size() > 0) {
+            predicate.setCcaPositions(ccaPositions);
         }
 
-        String[] fieldKeywords = trimmedArgs.split("\\s+");
+        if (!predicate.isFieldKeywordPresent()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    SearchCommand.MESSAGE_NO_KEYWORDS));
+        }
```

Comment:
Doing `search` alone in the command box doesn't return the usage format. Should be:
``` java
if (!predicate.isFieldKeywordPresent()) {
    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
            SearchCommand.MESSAGE_NO_KEYWORDS + "\n" + SearchCommand.MESSAGE_USAGE));
}
```